### ISSUE TITLE
The "flux get -w" watch command formats output inconsistently

### ISSUE THREAD

**Issue by kingdonb** [fluxcd/flux2#1848]
### Describe the bug

The output from `flux get ks` and `kubectl -n flux-system get ks` are consistently formatted, with each column getting the appropriate width so that all entries are lined up like a table.

The output from `flux get ks -w` and similar `get -w` commands does not take neighbors into account, so the width of each column is inconsistent from one line to the next. The output gets harder to read as the list grows more diverse:

```
$ flux get ks -w
NAME         	READY	MESSAGE                                                        	REVISION                                     	SUSPENDED
linkerd-infra	True 	Applied revision: main/59dcc73d82730fe60892c471aa52cef87b886bfd	main/59dcc73d82730fe60892c471aa52cef87b886bfd	False
local-path-provisioner	True	Applied revision: main/59dcc73d82730fe60892c471aa52cef87b886bfd	main/59dcc73d82730fe60892c471aa52cef87b886bfd	False
deis	True	Applied revision: main/59dcc73d82730fe60892c471aa52cef87b886bfd	main/59dcc73d82730fe60892c471aa52cef87b886bfd	False
deis-hosts	True	Applied revision: main/04b4193b6ee5a7d0591f87be55d78622a8d48c37	main/04b4193b6ee5a7d0591f87be55d78622a8d48c37	False
jenkins	True	Applied revision: main/59dcc73d82730fe60892c471aa52cef87b886bfd	main/59dcc73d82730fe60892c471aa52cef87b886bfd	False
keycloak	True	Applied revision: main/59dcc73d82730fe60892c471aa52cef87b886bfd	main/59dcc73d82730fe60892c471aa52cef87b886bfd	False
openvpn	True	Applied revision: main/59dcc73d82730fe60892c471aa52cef87b886bfd	main/59dcc73d82730fe60892c471aa52cef87b886bfd	False
persistence	True	Applied revision: main/59dcc73d82730fe60892c471aa52cef87b886bfd	main/59dcc73d82730fe60892c471aa52cef87b886bfd	False
linkerd-viz	True	Applied revision: main/04b4193b6ee5a7d0591f87be55d78622a8d48c37	main/04b4193b6ee5a7d0591f87be55d78622a8d48c37	False
metallb	True	Applied revision: main/59dcc73d82730fe60892c471aa52cef87b886bfd	main/59dcc73d82730fe60892c471aa52cef87b886bfd	False
metallb-instance	True	Applied revision: main/59dcc73d82730fe60892c471aa52cef87b886bfd	main/59dcc73d82730fe60892c471aa52cef87b886bfd	False
metrics-server	True	Applied revision: main/59dcc73d82730fe60892c471aa52cef87b886bfd	main/59dcc73d82730fe60892c471aa52cef87b886bfd	False
ambassador	True	Applied revision: main/04b4193b6ee5a7d0591f87be55d78622a8d48c37	main/04b4193b6ee5a7d0591f87be55d78622a8d48c37	False
cert-manager	True	Applied revision: main/59dcc73d82730fe60892c471aa52cef87b886bfd	main/59dcc73d82730fe60892c471aa52cef87b886bfd	False
cert-manager-issuers	True	Applied revision: main/59dcc73d82730fe60892c471aa52cef87b886bfd	main/59dcc73d82730fe60892c471aa52cef87b886bfd	False
flagger	True	Applied revision: main/04b4193b6ee5a7d0591f87be55d78622a8d48c37	main/04b4193b6ee5a7d0591f87be55d78622a8d48c37	False
podinfo	True	Applied revision: main/04b4193b6ee5a7d0591f87be55d78622a8d48c37	main/04b4193b6ee5a7d0591f87be55d78622a8d48c37	False
ingress-nginx	True	Applied revision: main/59dcc73d82730fe60892c471aa52cef87b886bfd	main/59dcc73d82730fe60892c471aa52cef87b886bfd	False
kube-oidc-proxy	True	Applied revision: main/59dcc73d82730fe60892c471aa52cef87b886bfd	main/59dcc73d82730fe60892c471aa52cef87b886bfd	False
linkerd	True	Applied revision: main/04b4193b6ee5a7d0591f87be55d78622a8d48c37	main/04b4193b6ee5a7d0591f87be55d78622a8d48c37	False
my-secrets	True	Applied revision: main/59dcc73d82730fe60892c471aa52cef87b886bfd	main/59dcc73d82730fe60892c471aa52cef87b886bfd	False
booksapp	True	Applied revision: main/04b4193b6ee5a7d0591f87be55d78622a8d48c37	main/04b4193b6ee5a7d0591f87be55d78622a8d48c37	False
emojivoto	True	Applied revision: main/04b4193b6ee5a7d0591f87be55d78622a8d48c37	main/04b4193b6ee5a7d0591f87be55d78622a8d48c37	False
flux-system	True	Applied revision: main/59dcc73d82730fe60892c471aa52cef87b886bfd	main/59dcc73d82730fe60892c471aa52cef87b886bfd	False
flux-system-extras	True	Applied revision: main/59dcc73d82730fe60892c471aa52cef87b886bfd	main/59dcc73d82730fe60892c471aa52cef87b886bfd	False
```

Compare to
```
$ kff get ks -w
NAME                     READY   STATUS                                                            AGE
ambassador               True    Applied revision: main/04b4193b6ee5a7d0591f87be55d78622a8d48c37   14h
booksapp                 True    Applied revision: main/04b4193b6ee5a7d0591f87be55d78622a8d48c37   2d19h
cert-manager             True    Applied revision: main/59dcc73d82730fe60892c471aa52cef87b886bfd   2d19h
cert-manager-issuers     True    Applied revision: main/59dcc73d82730fe60892c471aa52cef87b886bfd   2d19h
deis                     True    Applied revision: main/59dcc73d82730fe60892c471aa52cef87b886bfd   2d19h
deis-hosts               True    Applied revision: main/04b4193b6ee5a7d0591f87be55d78622a8d48c37   2d19h
emojivoto                True    Applied revision: main/04b4193b6ee5a7d0591f87be55d78622a8d48c37   2d19h
flagger                  True    Applied revision: main/04b4193b6ee5a7d0591f87be55d78622a8d48c37   2d19h
flux-system              True    Applied revision: main/59dcc73d82730fe60892c471aa52cef87b886bfd   2d19h
flux-system-extras       True    Applied revision: main/59dcc73d82730fe60892c471aa52cef87b886bfd   2d19h
ingress-nginx            True    Applied revision: main/59dcc73d82730fe60892c471aa52cef87b886bfd   2d19h
jenkins                  True    Applied revision: main/59dcc73d82730fe60892c471aa52cef87b886bfd   2d19h
keycloak                 True    Applied revision: main/59dcc73d82730fe60892c471aa52cef87b886bfd   2d19h
kube-oidc-proxy          True    Applied revision: main/59dcc73d82730fe60892c471aa52cef87b886bfd   2d19h
linkerd                  True    Applied revision: main/04b4193b6ee5a7d0591f87be55d78622a8d48c37   2d19h
linkerd-infra            True    Applied revision: main/59dcc73d82730fe60892c471aa52cef87b886bfd   2d19h
linkerd-viz              True    Applied revision: main/04b4193b6ee5a7d0591f87be55d78622a8d48c37   2d19h
local-path-provisioner   True    Applied revision: main/59dcc73d82730fe60892c471aa52cef87b886bfd   2d19h
metallb                  True    Applied revision: main/59dcc73d82730fe60892c471aa52cef87b886bfd   2d19h
metallb-instance         True    Applied revision: main/59dcc73d82730fe60892c471aa52cef87b886bfd   2d19h
metrics-server           True    Applied revision: main/59dcc73d82730fe60892c471aa52cef87b886bfd   2d19h
my-secrets               True    Applied revision: main/59dcc73d82730fe60892c471aa52cef87b886bfd   2d19h
openvpn                  True    Applied revision: main/59dcc73d82730fe60892c471aa52cef87b886bfd   2d19h
persistence              True    Applied revision: main/59dcc73d82730fe60892c471aa52cef87b886bfd   2d19h
podinfo                  True    Applied revision: main/04b4193b6ee5a7d0591f87be55d78622a8d48c37   2d19h
```

### Steps to reproduce

Start using Flux, add some non-trivial number of Kustomizations, and induce some of them to reconcile (with or without an error...)

```
kubectl get kustomizations -w
for i in $(kff get ks -o name|awk -F/ '{print $2}'); do flux reconcile ks $i; done
```

The output bounces back and forth between column widths (the output gets worse if there are long error messages displayed, but this minimal example is representative of the problem):

```
$ flux get ks -w
NAME   	READY	MESSAGE                                                        	REVISION                                     	SUSPENDED
openvpn	True 	Applied revision: main/59dcc73d82730fe60892c471aa52cef87b886bfd	main/59dcc73d82730fe60892c471aa52cef87b886bfd	False
persistence	True	Applied revision: main/59dcc73d82730fe60892c471aa52cef87b886bfd	main/59dcc73d82730fe60892c471aa52cef87b886bfd	False
ambassador	True	Applied revision: main/04b4193b6ee5a7d0591f87be55d78622a8d48c37	main/04b4193b6ee5a7d0591f87be55d78622a8d48c37	False
cert-manager	True	Applied revision: main/59dcc73d82730fe60892c471aa52cef87b886bfd	main/59dcc73d82730fe60892c471aa52cef87b886bfd	False
cert-manager-issuers	True	Applied revision: main/59dcc73d82730fe60892c471aa52cef87b886bfd	main/59dcc73d82730fe60892c471aa52cef87b886bfd	False
flagger	True	Applied revision: main/04b4193b6ee5a7d0591f87be55d78622a8d48c37	main/04b4193b6ee5a7d0591f87be55d78622a8d48c37	False
linkerd-viz	True	Applied revision: main/04b4193b6ee5a7d0591f87be55d78622a8d48c37	main/04b4193b6ee5a7d0591f87be55d78622a8d48c37	False
metallb	True	Applied revision: main/59dcc73d82730fe60892c471aa52cef87b886bfd	main/59dcc73d82730fe60892c471aa52cef87b886bfd	False
metallb-instance	True	Applied revision: main/59dcc73d82730fe60892c471aa52cef87b886bfd	main/59dcc73d82730fe60892c471aa52cef87b886bfd	False
metrics-server	True	Applied revision: main/59dcc73d82730fe60892c471aa52cef87b886bfd	main/59dcc73d82730fe60892c471aa52cef87b886bfd	False
podinfo	True	Applied revision: main/04b4193b6ee5a7d0591f87be55d78622a8d48c37	main/04b4193b6ee5a7d0591f87be55d78622a8d48c37	False
booksapp	True	Applied revision: main/04b4193b6ee5a7d0591f87be55d78622a8d48c37	main/04b4193b6ee5a7d0591f87be55d78622a8d48c37	False
emojivoto	True	Applied revision: main/04b4193b6ee5a7d0591f87be55d78622a8d48c37	main/04b4193b6ee5a7d0591f87be55d78622a8d48c37	False
flux-system	True	Applied revision: main/59dcc73d82730fe60892c471aa52cef87b886bfd	main/59dcc73d82730fe60892c471aa52cef87b886bfd	False
flux-system-extras	True	Applied revision: main/59dcc73d82730fe60892c471aa52cef87b886bfd	main/59dcc73d82730fe60892c471aa52cef87b886bfd	False
ingress-nginx	True	Applied revision: main/59dcc73d82730fe60892c471aa52cef87b886bfd	main/59dcc73d82730fe60892c471aa52cef87b886bfd	False
kube-oidc-proxy	True	Applied revision: main/59dcc73d82730fe60892c471aa52cef87b886bfd	main/59dcc73d82730fe60892c471aa52cef87b886bfd	False
linkerd	True	Applied revision: main/04b4193b6ee5a7d0591f87be55d78622a8d48c37	main/04b4193b6ee5a7d0591f87be55d78622a8d48c37	False
my-secrets	True	Applied revision: main/59dcc73d82730fe60892c471aa52cef87b886bfd	main/59dcc73d82730fe60892c471aa52cef87b886bfd	False
deis	True	Applied revision: main/59dcc73d82730fe60892c471aa52cef87b886bfd	main/59dcc73d82730fe60892c471aa52cef87b886bfd	False
deis-hosts	True	Applied revision: main/04b4193b6ee5a7d0591f87be55d78622a8d48c37	main/04b4193b6ee5a7d0591f87be55d78622a8d48c37	False
jenkins	True	Applied revision: main/59dcc73d82730fe60892c471aa52cef87b886bfd	main/59dcc73d82730fe60892c471aa52cef87b886bfd	False
keycloak	True	Applied revision: main/59dcc73d82730fe60892c471aa52cef87b886bfd	main/59dcc73d82730fe60892c471aa52cef87b886bfd	False
linkerd-infra	True	Applied revision: main/59dcc73d82730fe60892c471aa52cef87b886bfd	main/59dcc73d82730fe60892c471aa52cef87b886bfd	False
local-path-provisioner	True	Applied revision: main/59dcc73d82730fe60892c471aa52cef87b886bfd	main/59dcc73d82730fe60892c471aa52cef87b886bfd	False
ambassador	True	Applied revision: main/04b4193b6ee5a7d0591f87be55d78622a8d48c37	main/04b4193b6ee5a7d0591f87be55d78622a8d48c37	False
ambassador	Unknown	reconciliation in progress	main/04b4193b6ee5a7d0591f87be55d78622a8d48c37	False
ambassador	True	Applied revision: main/04b4193b6ee5a7d0591f87be55d78622a8d48c37	main/04b4193b6ee5a7d0591f87be55d78622a8d48c37	False
booksapp	True	Applied revision: main/04b4193b6ee5a7d0591f87be55d78622a8d48c37	main/04b4193b6ee5a7d0591f87be55d78622a8d48c37	False
booksapp	Unknown	reconciliation in progress	main/04b4193b6ee5a7d0591f87be55d78622a8d48c37	False
booksapp	True	Applied revision: main/04b4193b6ee5a7d0591f87be55d78622a8d48c37	main/04b4193b6ee5a7d0591f87be55d78622a8d48c37	False
cert-manager	True	Applied revision: main/59dcc73d82730fe60892c471aa52cef87b886bfd	main/59dcc73d82730fe60892c471aa52cef87b886bfd	False
cert-manager	Unknown	reconciliation in progress	main/59dcc73d82730fe60892c471aa52cef87b886bfd	False
cert-manager	True	Applied revision: main/59dcc73d82730fe60892c471aa52cef87b886bfd	main/59dcc73d82730fe60892c471aa52cef87b886bfd	False
cert-manager-issuers	True	Applied revision: main/59dcc73d82730fe60892c471aa52cef87b886bfd	main/59dcc73d82730fe60892c471aa52cef87b886bfd	False
cert-manager-issuers	Unknown	reconciliation in progress	main/59dcc73d82730fe60892c471aa52cef87b886bfd	False
cert-manager-issuers	True	Applied revision: main/59dcc73d82730fe60892c471aa52cef87b886bfd	main/59dcc73d82730fe60892c471aa52cef87b886bfd	False
deis	True	Applied revision: main/59dcc73d82730fe60892c471aa52cef87b886bfd	main/59dcc73d82730fe60892c471aa52cef87b886bfd	False
deis	Unknown	reconciliation in progress	main/59dcc73d82730fe60892c471aa52cef87b886bfd	False
deis	True	Applied revision: main/59dcc73d82730fe60892c471aa52cef87b886bfd	main/59dcc73d82730fe60892c471aa52cef87b886bfd	False
deis-hosts	True	Applied revision: main/04b4193b6ee5a7d0591f87be55d78622a8d48c37	main/04b4193b6ee5a7d0591f87be55d78622a8d48c37	False
deis-hosts	Unknown	reconciliation in progress	main/04b4193b6ee5a7d0591f87be55d78622a8d48c37	False
deis-hosts	True	Applied revision: main/04b4193b6ee5a7d0591f87be55d78622a8d48c37	main/04b4193b6ee5a7d0591f87be55d78622a8d48c37	False
emojivoto	True	Applied revision: main/04b4193b6ee5a7d0591f87be55d78622a8d48c37	main/04b4193b6ee5a7d0591f87be55d78622a8d48c37	False
emojivoto	Unknown	reconciliation in progress	main/04b4193b6ee5a7d0591f87be55d78622a8d48c37	False
emojivoto	True	Applied revision: main/04b4193b6ee5a7d0591f87be55d78622a8d48c37	main/04b4193b6ee5a7d0591f87be55d78622a8d48c37	False
flagger	True	Applied revision: main/04b4193b6ee5a7d0591f87be55d78622a8d48c37	main/04b4193b6ee5a7d0591f87be55d78622a8d48c37	False
flagger	Unknown	reconciliation in progress	main/04b4193b6ee5a7d0591f87be55d78622a8d48c37	False
flagger	True	Applied revision: main/04b4193b6ee5a7d0591f87be55d78622a8d48c37	main/04b4193b6ee5a7d0591f87be55d78622a8d48c37	False
flux-system	True	Applied revision: main/59dcc73d82730fe60892c471aa52cef87b886bfd	main/59dcc73d82730fe60892c471aa52cef87b886bfd	False
flux-system	Unknown	reconciliation in progress	main/59dcc73d82730fe60892c471aa52cef87b886bfd	False
flux-system	True	Applied revision: main/59dcc73d82730fe60892c471aa52cef87b886bfd	main/59dcc73d82730fe60892c471aa52cef87b886bfd	False
flux-system-extras	True	Applied revision: main/59dcc73d82730fe60892c471aa52cef87b886bfd	main/59dcc73d82730fe60892c471aa52cef87b886bfd	False
flux-system-extras	Unknown	reconciliation in progress	main/59dcc73d82730fe60892c471aa52cef87b886bfd	False
flux-system-extras	True	Applied revision: main/59dcc73d82730fe60892c471aa52cef87b886bfd	main/59dcc73d82730fe60892c471aa52cef87b886bfd	False
ingress-nginx	True	Applied revision: main/59dcc73d82730fe60892c471aa52cef87b886bfd	main/59dcc73d82730fe60892c471aa52cef87b886bfd	False
ingress-nginx	Unknown	reconciliation in progress	main/59dcc73d82730fe60892c471aa52cef87b886bfd	False
ingress-nginx	True	Applied revision: main/59dcc73d82730fe60892c471aa52cef87b886bfd	main/59dcc73d82730fe60892c471aa52cef87b886bfd	False
jenkins	True	Applied revision: main/59dcc73d82730fe60892c471aa52cef87b886bfd	main/59dcc73d82730fe60892c471aa52cef87b886bfd	False
jenkins	Unknown	reconciliation in progress	main/59dcc73d82730fe60892c471aa52cef87b886bfd	False
jenkins	True	Applied revision: main/59dcc73d82730fe60892c471aa52cef87b886bfd	main/59dcc73d82730fe60892c471aa52cef87b886bfd	False
local-path-provisioner	True	Applied revision: main/59dcc73d82730fe60892c471aa52cef87b886bfd	main/59dcc73d82730fe60892c471aa52cef87b886bfd	False
local-path-provisioner	Unknown	reconciliation in progress	main/59dcc73d82730fe60892c471aa52cef87b886bfd	False
local-path-provisioner	True	Applied revision: main/59dcc73d82730fe60892c471aa52cef87b886bfd	main/59dcc73d82730fe60892c471aa52cef87b886bfd	False
metallb	True	Applied revision: main/59dcc73d82730fe60892c471aa52cef87b886bfd	main/59dcc73d82730fe60892c471aa52cef87b886bfd	False
metallb	Unknown	reconciliation in progress	main/59dcc73d82730fe60892c471aa52cef87b886bfd	False
metallb	True	Applied revision: main/59dcc73d82730fe60892c471aa52cef87b886bfd	main/59dcc73d82730fe60892c471aa52cef87b886bfd	False
metallb-instance	True	Applied revision: main/59dcc73d82730fe60892c471aa52cef87b886bfd	main/59dcc73d82730fe60892c471aa52cef87b886bfd	False
metallb-instance	Unknown	reconciliation in progress	main/59dcc73d82730fe60892c471aa52cef87b886bfd	False
metallb-instance	True	Applied revision: main/59dcc73d82730fe60892c471aa52cef87b886bfd	main/59dcc73d82730fe60892c471aa52cef87b886bfd	False
metrics-server	True	Applied revision: main/59dcc73d82730fe60892c471aa52cef87b886bfd	main/59dcc73d82730fe60892c471aa52cef87b886bfd	False
metrics-server	Unknown	reconciliation in progress	main/59dcc73d82730fe60892c471aa52cef87b886bfd	False
metrics-server	True	Applied revision: main/59dcc73d82730fe60892c471aa52cef87b886bfd	main/59dcc73d82730fe60892c471aa52cef87b886bfd	False
my-secrets	True	Applied revision: main/59dcc73d82730fe60892c471aa52cef87b886bfd	main/59dcc73d82730fe60892c471aa52cef87b886bfd	False
my-secrets	Unknown	reconciliation in progress	main/59dcc73d82730fe60892c471aa52cef87b886bfd	False
my-secrets	True	Applied revision: main/59dcc73d82730fe60892c471aa52cef87b886bfd	main/59dcc73d82730fe60892c471aa52cef87b886bfd	False
openvpn	True	Applied revision: main/59dcc73d82730fe60892c471aa52cef87b886bfd	main/59dcc73d82730fe60892c471aa52cef87b886bfd	False
openvpn	Unknown	reconciliation in progress	main/59dcc73d82730fe60892c471aa52cef87b886bfd	False
openvpn	True	Applied revision: main/59dcc73d82730fe60892c471aa52cef87b886bfd	main/59dcc73d82730fe60892c471aa52cef87b886bfd	False
persistence	True	Applied revision: main/59dcc73d82730fe60892c471aa52cef87b886bfd	main/59dcc73d82730fe60892c471aa52cef87b886bfd	False
persistence	Unknown	reconciliation in progress	main/59dcc73d82730fe60892c471aa52cef87b886bfd	False
persistence	True	Applied revision: main/59dcc73d82730fe60892c471aa52cef87b886bfd	main/59dcc73d82730fe60892c471aa52cef87b886bfd	False
podinfo	True	Applied revision: main/04b4193b6ee5a7d0591f87be55d78622a8d48c37	main/04b4193b6ee5a7d0591f87be55d78622a8d48c37	False
podinfo	Unknown	reconciliation in progress	main/04b4193b6ee5a7d0591f87be55d78622a8d48c37	False
podinfo	True	Applied revision: main/04b4193b6ee5a7d0591f87be55d78622a8d48c37	main/04b4193b6ee5a7d0591f87be55d78622a8d48c37	False
flux-system	Unknown	reconciliation in progress	main/59dcc73d82730fe60892c471aa52cef87b886bfd	False
flux-system	True	Applied revision: main/59dcc73d82730fe60892c471aa52cef87b886bfd	main/59dcc73d82730fe60892c471aa52cef87b886bfd	False
```

There are no columns. Output is hard to follow from one line to the next.

### Expected behavior

The output should be formatted in columns, which expand as needed and stay that wide for the duration of the output, ideally the same way that `kubectl get -w` does it:

```
$ kubectl -n flux-system get ks -w
NAME                     READY   STATUS                                                            AGE
ambassador               True    Applied revision: main/04b4193b6ee5a7d0591f87be55d78622a8d48c37   14h
booksapp                 True    Applied revision: main/04b4193b6ee5a7d0591f87be55d78622a8d48c37   2d19h
...
local-path-provisioner   True    Applied revision: main/59dcc73d82730fe60892c471aa52cef87b886bfd   2d19h
metallb                  True    Applied revision: main/59dcc73d82730fe60892c471aa52cef87b886bfd   2d19h
metallb-instance         True    Applied revision: main/59dcc73d82730fe60892c471aa52cef87b886bfd   2d19h
metrics-server           True    Applied revision: main/59dcc73d82730fe60892c471aa52cef87b886bfd   2d19h
my-secrets               True    Applied revision: main/59dcc73d82730fe60892c471aa52cef87b886bfd   2d19h
openvpn                  True    Applied revision: main/59dcc73d82730fe60892c471aa52cef87b886bfd   2d19h
persistence              True    Applied revision: main/59dcc73d82730fe60892c471aa52cef87b886bfd   2d19h
podinfo                  True    Applied revision: main/04b4193b6ee5a7d0591f87be55d78622a8d48c37   2d19h
ambassador               True    Applied revision: main/04b4193b6ee5a7d0591f87be55d78622a8d48c37   14h
ambassador               Unknown   reconciliation in progress                                        14h
...
linkerd-viz              Unknown   reconciliation in progress                                        2d19h
linkerd-viz              True      Applied revision: main/04b4193b6ee5a7d0591f87be55d78622a8d48c37   2d19h
local-path-provisioner   True      Applied revision: main/59dcc73d82730fe60892c471aa52cef87b886bfd   2d19h
local-path-provisioner   Unknown   reconciliation in progress                                        2d19h
local-path-provisioner   True      Applied revision: main/59dcc73d82730fe60892c471aa52cef87b886bfd   2d19h
metallb                  True      Applied revision: main/59dcc73d82730fe60892c471aa52cef87b886bfd   2d19h
metallb                  Unknown   reconciliation in progress                                        2d19h
metallb                  True      Applied revision: main/59dcc73d82730fe60892c471aa52cef87b886bfd   2d19h
metallb-instance         True      Applied revision: main/59dcc73d82730fe60892c471aa52cef87b886bfd   2d19h
metallb-instance         Unknown   reconciliation in progress                                        2d19h
metallb-instance         True      Applied revision: main/59dcc73d82730fe60892c471aa52cef87b886bfd   2d19h
...
metallb                  Unknown   reconciliation in progress                                        2d19h
linkerd-infra            True      Applied revision: main/59dcc73d82730fe60892c471aa52cef87b886bfd   2d19h
linkerd-infra            False     dependency 'flux-system/openvpn' is not ready                     2d19h
openvpn                  True      Applied revision: main/59dcc73d82730fe60892c471aa52cef87b886bfd   2d19h
metallb-instance         True      Applied revision: main/59dcc73d82730fe60892c471aa52cef87b886bfd   2d19h
metallb-instance         False     dependency 'flux-system/metallb' is not ready                     2d19h
metallb                  True      Applied revision: main/59dcc73d82730fe60892c471aa52cef87b886bfd   2d19h
```

Each column has a "high-water mark" width and as the output in that column grows wider, the terminal column width scales up to accommodate the widest output that has been in that column so far, but never scales back down for the duration of the run.

Note: I am not sure this is the best strategy to ensure readable output, as it still doesn't work very well when the error output is long, ...but it definitely results in IMHO much more readable output over the execution of the watch command.

### OS / Distro

MacOS 10.15.7

### Flux version

0.17.2

### Flux check

$ flux check
► checking prerequisites
✔ kubectl 1.22.1 >=1.18.0-0
✔ Kubernetes 1.21.4 >=1.16.0-0
► checking controllers
✔ helm-controller: deployment ready
► ghcr.io/fluxcd/helm-controller:v0.11.2
✔ image-automation-controller: deployment ready
► ghcr.io/fluxcd/image-automation-controller:v0.14.1
✔ image-reflector-controller: deployment ready
► ghcr.io/fluxcd/image-reflector-controller:v0.11.1
✔ kustomize-controller: deployment ready
► ghcr.io/fluxcd/kustomize-controller:v0.14.1
✔ notification-controller: deployment ready
► ghcr.io/fluxcd/notification-controller:v0.16.0
✔ source-controller: deployment ready
► ghcr.io/fluxcd/source-controller:v0.15.4
✔ all checks passed

### Code of Conduct

- [X] I agree to follow this project's Code of Conduct

**Comment by kingdonb** [comment]
I'm no longer 100% convinced we should adopt the behavior of kubectl

I found at least one case where `kubectl get ks -w` produces unusable output and `flux get ks -w` is better

Leaving this open for documentation purposes, since someone might still not agree? But if there is anyone who was planning to follow up this issue on my account, I would say that it can be closed.

**Comment by SomtochiAma** [comment]
Hey @kingdonb , I am looking into this issue. Can you also give the instance where `kubectl get ks -w` produces unusable output  `flux get ks -w` is better?

**Comment by yebyen** [comment]
Sure! My repo is https://github.com/kingdon-ci/jenkins-infra, the secrets all need to be rotated to come up, I need to add some instructions (before Kubecon) about how to do all that... I'll put some instructions together soon!

**Comment by yebyen** [comment]
Basically any instance where there are this many Kustomization and some of them have long errors makes this (the kubectl behavior) unusable. You can only see about 4 or 5 entries per terminal page. I noticed the error messages in v1beta2 Kustomize controller are a bit neater and generally good experience with that one so far!
